app/models/concerns/episode_media.rb

```diff
@@ -3,19 +3,32 @@
 module EpisodeMedia
   extend ActiveSupport::Concern
 
+  def cut_media_version!
+    latest_version = media_versions.first
+    latest_media = latest_version&.media_resources || []
+    latest_ids = latest_media.map(&:id)
+
+    # backfill media_versions for newly completed media
+    if media_ready? && latest_ids != media_ids
+      new_version = media_versions.build
+      media.each { |m| new_version.media_version_resources.build(media_resource: m) }
+      new_version.save!
+      new_version
+    else
+      latest_version
+    end
+  end
```

Comment:
Okay, so creating media versions...

I landed on _not_ init-ing these on save.  They're only created once all media files complete processing, they're the correct mime type, and they match the segment count.  This is a bit more SQL to run, but it seemed safest to fill them in as we publish the version of the media out.

The complete flow is:

1. You upload/save `episode.contents` (or we create them by segmenting the `episode.uncut`).
2. Async processing kicks off - eventually all the files complete.
3. If/when the publish jobs (apple or RSS) run, they eventually generate the eps in the feed, and ask [ep.is_include_in_feed?](https://github.com/PRX/feeder.prx.org/blob/main/app/models/feed.rb#L125)
4. If the episode purposefully has no media, it's included in the feed.  And there's no media_version to create.
5. If it has media, we call `complete_media?` (below) - which in turn calls this method.
6. This method checks if we need to create a new version or not.

And that's it.  Fairly simple?  Maybe a bit magical how we get here?  But I've thought/tested/QA'd through a bunch of edge cases, where you're swapping files before we publish the prior version.  And this holds up.  Since we're always (soft-) deleting the old files when we replace them.  And all of publishing is gated on `complete_media?`.

Anyways, any thoughts on that?  Ways to make it clearer what's happening?